student-projects/gregory/Sdutents-Stats/app.ts

```diff
@@ -0,0 +1,59 @@
+let StatType = 0;
+let Answer = "";
+while (!((StatType === 1) || (StatType === 2)))  {
+    Answer = prompt("Student Statistics. Do you want '1' - Best Grade or '2' - Worst Grade?");
+    if (Answer === null) {
+        alert("Looks like you don't want any grade. Good Day!");
+        throw "Canceled by the user.";
+    }
+    StatType = Number(Answer);
+    if (!((StatType === 1) || (StatType === 2))){
+        alert("ERROR - wrong input, it must be 1 or 2!");    
+    } 
+}
+
+let MinGrade = 100;
+let MaxGrade = 0;
+let StudentName = "";
+let NameAnswer = "";
+while (Answer !== null){
```

Comment:
Don't reuse variables for different purposes.
It's clearer to have variables with specific tasks, like `statTypeInput`, `currentStudentName` and `currentGrade`